lib/engine/src/artifact.rs

```diff
@@ -161,7 +161,7 @@ pub trait Artifact: Send + Sync + Upcastable + MemoryUsage {
     /// See [`InstanceHandle::finish_instantiation`].
     unsafe fn finish_instantiation(
         &self,
-        trap_handler: &dyn TrapHandler,
+        trap_handler: Option<*const TrapHandler<'static>>,
```

Comment:
Just a note about style, it's a little bit odd to see `Option<*const T>` as this is essentially a doubly nullable pointer. I think I'd prefer using `Option<NonNull<T>>` here as that uses the null representation of the pointer itself as the `Option::None` variant. Not a huge concern though

Reply:
Alternatively, when used in the argument position, I think it's fine to just use `*const TrapHandler<'static>` here too. And using the null representation instead of `Option`. I like `Option` because it forces the receiver to check if it's null though

Reply:
Used `Arc` at the end :)

Reply:
:+1: